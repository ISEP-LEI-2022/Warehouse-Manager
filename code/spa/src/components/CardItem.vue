<template>
  <div class="col-12 lg:col-6 xl:col-3">
    <div class="card mb-0">
      <div class="flex justify-content-between mb-3">
        <div :id="`${props.title}-card-item`">
          <span class="block text-500 font-medium mb-3">{{ props.title }}</span>
          <div class="text-900 font-medium text-xl">{{ props.content }}</div>
        </div>
        <IconElement :icon="props.title" />
      </div>
    </div>
  </div>
</template>
<script setup lang="ts">
const props = defineProps<{
  title: string;
  content: string;
}>();
</script>
