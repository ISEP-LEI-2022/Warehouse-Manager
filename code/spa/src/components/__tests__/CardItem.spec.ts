import { describe, it, expect } from 'vitest'
import { mount } from '@vue/test-utils'
import CardItem from '../CardItem.vue'


describe('CrudDialog', () => {
  it('renders properly', () => {
    const wrapper = mount(CardItem,
      {
        props:
        {
          title: "CardItem test",
          content: "Hello this is a CardItem test",
        }
    }
    )
    expect(wrapper.text()).toMatchSnapshot()
  })
})
