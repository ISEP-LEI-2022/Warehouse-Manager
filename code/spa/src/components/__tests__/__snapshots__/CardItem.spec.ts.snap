// Vitest Snapshot v1

exports[`CrudDialog > renders properly 1`] = `"CardItem testHello this is a CardItem test"`;
